lib/server.js

```diff
@@ -25,14 +25,24 @@ const {
 } = require('./management/agentClient');
 
 const routes = arsenal.s3routes.routes;
+const { parseLC, MultipleBackendGateway } = arsenal.storage.data;
 const websiteEndpoints = _config.websiteEndpoints;
+let client = dataWrapper.client;
+const implName = dataWrapper.implName;
 
 let allEndpoints;
 function updateAllEndpoints() {
     allEndpoints = Object.keys(_config.restEndpoints);
 }
 _config.on('rest-endpoints-update', updateAllEndpoints);
 updateAllEndpoints();
+_config.on('location-constraints-update', () => {
+    if (implName === 'multipleBackends') {
+        const clients = parseLC(_config, vault);
+        client = new MultipleBackendGateway(
+            clients, metadata, locationStorageCheck);
```

Comment:
should call client setter here for `dataWrapper`?

Reply:
Yeah, our codepath for 'GETs' are a little different, and why we need another event handler. We're not actually instantiating a DataWrapper here because it would lose its scope reference for `this` being passed around through Arsenal, so the event handler is the only way to make sure the location constraint list is updated for the DataWrapper that's instantiated like 5 functions down the line here https://github.com/scality/Arsenal/blob/development/8.1/lib/s3routes/routesUtils.js#L325

Reply:
I see. Thanks for the explanation. I had overlooked that the `clients` in `server.js` and `data/wrapper.js` do not reference the same object.